traitlets/traitlets.py

```diff
@@ -872,6 +885,112 @@ def setup_instance(self):
         self._trait_validators = {}
         super(HasTraits, self).setup_instance()
 
+    @classmethod
+    def setup_class(cls):
+        cls._write_docs_to_class()
+        MetaHasTraits.setup_class(cls)
+
+    @classmethod
+    def _write_docs_to_class(cls):
+        if hasattr(cls.__init__, 'im_func'):
+            docs = cls._rst_trait_help() or ""
+            if docs:
+                # copy the image function
+                im = cls.__init__.im_func
+                n, c, d, g = (im.__name__, im.__code__,
+                              im.__defaults__, im.__globals__)
+                f = types.FunctionType(c, g, n, d, im.__closure__)
```

Comment:
I did this for two reasons:

1. I edit `im_func` because, as it turns out, in Python 2, `method.__doc__` isn't writable
2. I copy the image function because all subclass reference `cls.__init__`. Thus, because of 1, all subclasses would essentially be trying to edit the parent docstring.

So instead of contend with all that I've now decided to assign trait/event docs to `cls.__doc__`.